airflow/dag_processing/manager.py

```diff
@@ -526,13 +527,11 @@ def deactivate_stale_dags(
         Deactivate them and remove them in the serialized_dag table.
         """
         to_deactivate = set()
-        query = session.query(DagModel.dag_id, DagModel.fileloc, DagModel.last_parsed_time).filter(
-            DagModel.is_active
-        )
+        query = select(DagModel.dag_id, DagModel.fileloc, DagModel.last_parsed_time).where(DagModel.is_active)
         standalone_dag_processor = conf.getboolean("scheduler", "standalone_dag_processor")
         if standalone_dag_processor:
-            query = query.filter(DagModel.processor_subdir == dag_directory)
-        dags_parsed = query.all()
+            query = query.where(DagModel.processor_subdir == dag_directory)
+        dags_parsed = session.execute(query)
```

Comment:
Based on the migration doc, you need to add `.scalars().all()`
```suggestion
        dags_parsed = session.execute(query).scalars().all()
```

Reply:
The `.all()` part in the previous query is superfulous since `dags_parsed` is simply iterated over. So the new code does not need `.all()` either. The `.scalars()` part is not correct here since we are selecting multiple fields and do want a Row object instead of scalars.